pjlib/include/pj/argparse.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2008-2024 Teluu Inc. (http://www.teluu.com)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#ifndef __PJ_ARGPARSE_H__
+#define __PJ_ARGPARSE_H__
+
+/**
+ * @file argparse.h
+ * @brief Command line argument parser
+ */
+#include <pj/ctype.h>
+#include <pj/errno.h>
+#include <pj/string.h>
+
+PJ_BEGIN_DECL
+
+/**
+ * Define function to display parsing error.
+ */
+#ifndef PJ_ARGPARSE_ERROR
+#  include <stdio.h>
+#  define PJ_ARGPARSE_ERROR(fmt, arg) printf(fmt "\n", arg)
+#endif
+
+
+/**
+ * @defgroup PJ_ARGPARSE Command line argument parser
+ * @ingroup PJ_MISC
+ * @{
+ *
+ * This module provides header only utilities to parse command line arguments.
+ * This is mostly used by PJSIP test and sample apps. Note that there is
+ * getopt() implementation in PJLIB-UTIL (but it's in PJLIB-UTIL, so it can't
+ * be used by PJLIB)
+ */
+
+/**
+ * Peek the next possible option from argv. An argument is considered an
+ * option if it starts with "-" and followed by at least another letter that
+ * is not digit or starts with "--" and followed by a letter.
+ *
+ * @param argv      The argv, which must be null terminated.
+ *
+ * @return next option or NULL.
+ */
+PJ_INLINE(char*) pj_argparse_peek_next_option(char *const argv[])
+{
+    while (*argv) {
+        const char *arg = *argv;
+        if ((*arg=='-' && *(arg+1) && !pj_isdigit(*(arg+1))) ||
+            (*arg=='-' && *(arg+1)=='-' && *(arg+2)))
```

Comment:
according to the spec, *(arg+2) also needs to be a letter, so need to check `!pj_isdigit()` as well?

Reply:
The comment meant to say that a negative number (e.g. `-1`) will not be considered as an option. But thinking about it again, and checking the behavior of some programs e.g. `gcc` or `clang`, they both consider `-1` as an option, e.g.:

```
$ gcc -1
gcc: error: unrecognized command-line option ‘-1’
```

So in the latest commit, any arguments that start with dash is considered an option (and comment updated accordingly)